### PR TITLE
Rebuild msitools 0.100 with libxml2 2.10

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Actions:
1. Fix source/url
2. Apply a patch only on osx
3. Reorder dependencies
4. Add libxml2 pinning

### CODE REVIEW

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Remove this tag when the underlying issue https://github.com/anaconda-distribution/rocket-platform/pull/751
+# will be resolved.
+pkg_build_image_tag: '2023.03.24'
```
Comment (M-Waszkiewicz-Anaconda) [inline abs.yaml]: lets wait with that until new docker is live and kicking

File: recipe/meta.yaml [hunk 2]
```diff
@@ -6,38 +6,37 @@ package:
...
-    - m2-patch  # [win]
-    - perl
   host:
-    - libgsf 1.14.*
-    - libuuid <2
-    - libxml2
     - gcab
-    - glib
     - gettext  # [osx]
+    - glib
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:33]: missing pinning

File: recipe/meta.yaml [hunk 3]
```diff
@@ -6,38 +6,37 @@ package:
...
+    - libgsf 1.14.*
+    - libuuid <2
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: missing strict pinning

File: recipe/meta.yaml [hunk 4]
```diff
@@ -6,38 +6,37 @@ package:
...
-    - perl
   host:
-    - libgsf 1.14.*
-    - libuuid <2
-    - libxml2
     - gcab
-    - glib
     - gettext  # [osx]
+    - glib
+    - libgsf 1.14.*
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: missing strict pinning

File: abs.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply abs.yaml]: Yeah, right

File: recipe/meta.yaml (hunk 4 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Fixed

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Fixed

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:33]: glib has a mess of pinnings https://metayaml-conda.fly.dev/metayaml/reverse_dependencies?name=glib. I just leave it without pinning as it's from cbc.yaml

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by cbouss — APPROVED** [review]

**Pull request merged**